### PR TITLE
Fix #482 and cleanup Registrator.js

### LINKED ISSUE #482: Registrator.js should use REGISTER OK Expires header, if expires parameter is missed in REGISTER OK Contact.
Exists a lot of different SIP servers,
for example the server don't  set expires parameter in REGISTER OK Contact:

REGISTER sip:audiocodes.com SIP/2.0
Via: SIP/2.0/WSS n0490ata5bdn.invalid;branch=z9hG4bK286403
Max-Forwards: 69
To: <sip:web1@audiocodes.com>
From: "Igor" <sip:web1@audiocodes.com>;tag=ton4e0ko03
Call-ID: gliihla37ig1s7fu0g94vs
CSeq: 2 REGISTER
Authorization: Digest algorithm=MD5, username="web1", realm="DefaultSipRealm", nonce="MTU4MzA3NTI6MjA4MzU4MzgzMA==", uri="sip:audiocodes.com", response="a0f3340ea63a486746998c9a2f805e54", qop=auth-int, cnonce="lhtrjrdl96in", nc=00000001
Contact: <sip:447o544j@n0490ata5bdn.invalid;transport=ws>;+sip.ice;reg-id=1;+sip.instance="<urn:uuid:2be93214-4445-4132-bfb5-eb1b413bc909>";expires=600
Expires: 600
Allow: INVITE,ACK,CANCEL,BYE,UPDATE,MESSAGE,OPTIONS,REFER,INFO
Supported: path,gruu,outbound
User-Agent: JsSIP 3.1.2
Content-Length: 0

SIP/2.0 200 OK
Via: SIP/2.0/WSS n0490ata5bdn.invalid;branch=z9hG4bK286403
From: "Igor" <sip:web1@audiocodes.com>;tag=ton4e0ko03
To: <sip:web1@audiocodes.com>;tag=1c30910671
Call-ID: gliihla37ig1s7fu0g94vs
CSeq: 2 REGISTER
Contact: <sip:447o544j@n0490ata5bdn.invalid;transport=ws>;+sip.instance="<urn:uuid:2be93214-4445-4132-bfb5-eb1b413bc909>";+sip.ice;reg-id=1
Allow: REGISTER,OPTIONS,INVITE,ACK,CANCEL,BYE,NOTIFY,PRACK,REFER,INFO,SUBSCRIBE,UPDATE
Expires: 260
Server: Mediant VE SBC/v.7.20A-u2980.160.690
Content-Length: 0
SIP/2.0 200 OK
Via: SIP/2.0/WSS n0490ata5bdn.invalid;branch=z9hG4bK286403
From: "Igor" <sip:web1@audiocodes.com>;tag=ton4e0ko03
To: <sip:web1@audiocodes.com>;tag=1c30910671
Call-ID: gliihla37ig1s7fu0g94vs
CSeq: 2 REGISTER
Contact: <sip:447o544j@n0490ata5bdn.invalid;transport=ws>;+sip.instance="<urn:uuid:2be93214-4445-4132-bfb5-eb1b413bc909>";+sip.ice;reg-id=1
Allow: REGISTER,OPTIONS,INVITE,ACK,CANCEL,BYE,NOTIFY,PRACK,REFER,INFO,SUBSCRIBE,UPDATE
Expires: 260
Server: Mediant VE SBC/v.7.20A-u2980.160.690
Content-Length: 0

Next REGISTER should be send after 260 - 5 seconds.
But  jssip it will send after 600 - 5 seconds.

It's because misuse of OK expires header:

It set expires variable as value of expires header, 
but then replace it to undefined when try to get expires parameter from Contact header.
So will be used this._expires.
 
Should be: 
If REGISTER OK Contact has expires parameter - use it, 
otherwise use REGISTER OK expires header,
otherwise expires from REGISTER request.

Proposed fix:  please change Registrator.js code that receive 2xx response to:

         case /^2[0-9]{2}$/.test(response.status_code):
            this._registering = false;

            // Search the Contact pointing to us and update the expires value accordingly.
            if (!contacts)
            {
              debug('no Contact header in response to REGISTER, response ignored');
              break;
            }

            while (contacts--)
            {
              contact = response.parseHeader('contact', contacts);
              if (contact.uri.user === this._ua.contact.uri.user)
              {
                expires = contact.getParam('expires');
                break;
              }
              else
              {
                contact = null;
              }
            }

            if (!contact)
            {
              debug('no Contact header pointing to us, response ignored');
              break;
            }

            if (!expires && response.hasHeader('expires'))
            {
              expires = response.getHeader('expires');
            }

            if (!expires)
            {
              expires = this._expires;
            }
. . . . . . . . . . .

### PR DESCRIPTION
**Pull request by ibc**

### CODE REVIEW

File: lib/Registrator.js [hunk 1]
```diff
@@ -127,7 +129,9 @@ ${this._contact};expires=${this._expires}${this._extraContactParams}`);
       },
       onReceiveResponse : (response) =>
       {
-        let contact, expires, contacts = response.getHeaders('contact').length;
+        let contact;
+        let expires;
+        let numContacts = response.getHeaders('contact').length;
```
Comment (jmillan) [inline lib/Registrator.js]: Why the change of the name to `numContacts`?

Comment (jmillan) [reply lib/Registrator.js]: hehe, silly question.., I missed the `length` keyword.

Comment (ibc) [reply lib/Registrator.js]: yep, at first sight I thought `contacts` was an array.
